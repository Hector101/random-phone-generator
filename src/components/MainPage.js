import React from "react";
import ReactTable from "react-table";

import Modal from './Modal';
import Result from './Result';

import generateNumber from '../helpers/generateNumber';
import { Phone  } from '../assets';

import '../styles/MainPage.scss';

class MainPage extends React.Component {
  state = {
    data: [],
    error: true,
    number: null,
    show: false,
  };

  _handleChange = (event) => {
    const { value } = event.target;
    const disable = Number(value) < 1 || Number(value) > 10000 || !value;
    this.setState({
      number: Number(event.target.value),
      error: disable ? true : false,
    })
  };

  _generateNumbers = () => {
    const { number } = this.state;
    const generatedNumber = generateNumber(number);
    this.setState({
      data: generatedNumber,
      show: true,
    })
  };

  _closeModal = () => {
    this.setState({
      show: false,
    })
  };
  _openModal = () => {
    this.setState({
      show: true,
    })
  };

  render() {
    const { data, error, show, number } = this.state;
    return (
      <>
        <Modal>
          <Result
            show={show}
            _closeModal={this._closeModal}
            data={data}
            number={number}
          />
        </Modal>
        <div className="w-100 vh-100 bg-light-gray flex justify-center items-center">
          <div className="main_wrapper">
            <div className="w-100 h4 pa3 flex flex-column justify-center items-center bb b--light-gray">
              <Phone className="w3 h3 logo"/>
              <span className="f3 b black-80 mt2">Phone Number Generator</span>
            </div>
            <div className="main_container w-100">
              <div className="w-100 pv3 control flex justify-center item-center">
                <input
                  className="error input-number outline-0 w4 bw1 ba b--gray"
                  type="number"
                  min="1"
                  max="10000"
                  placeholder="Eg: 15"
                  name="number"
                  onChange={this._handleChange}
                />
                <button
                  className="generate bn outline-0 pointer ml1 white ph2 ttu fw3 f6"
                  onClick={this._generateNumbers}
                  disabled={error}
                >Generate</button>
                <button
                  className={`${!!data.length ? 'db' : 'dn'} outline-0 pointer ml1 ph2 ttu fw3 f6 ba bw2 b--blue`}
                  onClick={this._generateNumbers}
                >View Result</button>
              </div>
              <div className="table flex justify-center">
                <ReactTable
                  data={data}
                  defaultPageSize={10}
                  columns={[{
                    Header: 'ID',
                    accessor: 'id',
                    className: 'flex justify-start fw1',
                    headerClassName: 'flex justify-start b outline-0'
                  },
                  {
                    Header: 'Phone Numbers',
                    accessor: 'phoneNumber',
                    className: 'flex justify-end',
                    headerClassName: 'flex justify-end b outline-0'
                  }]}
                  noDataText={'Numbers Not Generated'}
                  className="-striped -highlight w-50 outline-0"
                />
              </div>
            </div>
          </div>
        </div>
      </>
    )
  };
};

export default MainPage;
