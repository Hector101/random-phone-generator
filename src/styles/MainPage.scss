.main_wrapper {
  width: 100%;
  max-width: 90%;
  background: #ffffff;
  height: 100vh;

  .logo {
    fill: rgb(84, 149, 224);
  }

  .main_container {
    height: calc(100vh - 8rem);

    .control {
      .error {

      }

      .input-number {
        -moz-appearance: textfield;
        
        &::-webkit-inner-spin-button, ::-webkit-outer-spin-button {
          -webkit-appearance: none;
          margin: 0;
        }
      }

      .generate {
        background: rgb(84, 149, 224);
      }

      .generate {
        &:disabled {
          background: rgb(163, 197, 236);
        }
      }
    }
  }
}