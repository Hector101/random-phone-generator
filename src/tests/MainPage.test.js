import React from 'react';
import { shallow } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';
import { spy } from 'sinon';

import MainPage from '../components/MainPage';

describe('MainPage', () => {
  const wrapper = shallow(<MainPage />);

  it('components mounted successfully', () => {
    expect(EnzymeToJson(wrapper)).toMatchSnapshot();
  });

  it('should show modal on click of _generateNumbers method', () => {
    wrapper.instance()._generateNumbers()
    expect(wrapper.state().show).toBe(true);
  });

  it('should hide modal on click of _closeModal method', () => {
    wrapper.instance()._closeModal()
    expect(wrapper.state().show).toBe(false);
  });

  it('should explicitly show modal on click of _openModal method', () => {
    wrapper.instance()._openModal()
    expect(wrapper.state().show).toBe(true);
  });

  it('input of value calls _handleChange method', () => {
    const spy = jest.spyOn(wrapper.instance(), '_handleChange');

    wrapper.find('input').simulate('change', { target: { value: '12' } });
  });
});
