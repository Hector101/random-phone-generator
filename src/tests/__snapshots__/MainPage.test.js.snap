// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage components mounted successfully 1`] = `
<Fragment>
  <Modal>
    <Result
      _closeModal={[Function]}
      data={Array []}
      number={null}
      show={false}
    />
  </Modal>
  <div
    className="w-100 vh-100 bg-light-gray flex justify-center items-center"
  >
    <div
      className="main_wrapper"
    >
      <div
        className="w-100 h4 pa3 flex flex-column justify-center items-center bb b--light-gray"
      >
        <Phone.svg
          className="w3 h3 logo"
        />
        <span
          className="f3 b black-80 mt2"
        >
          Phone Number Generator
        </span>
      </div>
      <div
        className="main_container w-100"
      >
        <div
          className="w-100 pv3 control flex justify-center item-center"
        >
          <input
            className="error input-number outline-0 w4 bw1 ba b--gray"
            max="10000"
            min="1"
            name="number"
            onChange={[Function]}
            placeholder="Eg: 15"
            type="number"
          />
          <button
            className="generate bn outline-0 pointer ml1 white ph2 ttu fw3 f6"
            disabled={true}
            onClick={[Function]}
          >
            Generate
          </button>
          <button
            className="dn outline-0 pointer ml1 ph2 ttu fw3 f6 ba bw2 b--blue"
            onClick={[Function]}
          >
            View Result
          </button>
        </div>
        <div
          className="table flex justify-center"
        >
          <ReactTable
            AggregatedComponent={[Function]}
            ExpanderComponent={[Function]}
            FilterComponent={[Function]}
            LoadingComponent={[Function]}
            NoDataComponent={[Function]}
            PadRowComponent={[Function]}
            PaginationComponent={[Function]}
            PivotValueComponent={[Function]}
            ResizerComponent={[Function]}
            TableComponent={[Function]}
            TbodyComponent={[Function]}
            TdComponent={[Function]}
            TfootComponent={[Function]}
            ThComponent={[Function]}
            TheadComponent={[Function]}
            TrComponent={[Function]}
            TrGroupComponent={[Function]}
            aggregatedKey="_aggregated"
            className="-striped -highlight w-50 outline-0"
            collapseOnDataChange={true}
            collapseOnPageChange={true}
            collapseOnSortingChange={true}
            column={
              Object {
                "Aggregated": undefined,
                "Cell": undefined,
                "Expander": undefined,
                "Filter": undefined,
                "Footer": undefined,
                "Header": undefined,
                "Pivot": undefined,
                "PivotValue": undefined,
                "Placeholder": undefined,
                "aggregate": undefined,
                "className": "",
                "filterAll": false,
                "filterMethod": undefined,
                "filterable": undefined,
                "footerClassName": "",
                "footerStyle": Object {},
                "getFooterProps": [Function],
                "getHeaderProps": [Function],
                "getProps": [Function],
                "headerClassName": "",
                "headerStyle": Object {},
                "minResizeWidth": 11,
                "minWidth": 100,
                "resizable": undefined,
                "show": true,
                "sortMethod": undefined,
                "sortable": undefined,
                "style": Object {},
              }
            }
            columns={
              Array [
                Object {
                  "Header": "ID",
                  "accessor": "id",
                  "className": "flex justify-start fw1",
                  "headerClassName": "flex justify-start b outline-0",
                },
                Object {
                  "Header": "Phone Numbers",
                  "accessor": "phoneNumber",
                  "className": "flex justify-end",
                  "headerClassName": "flex justify-end b outline-0",
                },
              ]
            }
            data={Array []}
            defaultExpanded={Object {}}
            defaultFilterMethod={[Function]}
            defaultFiltered={Array []}
            defaultPage={0}
            defaultPageSize={10}
            defaultResized={Array []}
            defaultSortDesc={false}
            defaultSortMethod={[Function]}
            defaultSorted={Array []}
            expanderDefaults={
              Object {
                "filterable": false,
                "resizable": false,
                "sortable": false,
                "width": 35,
              }
            }
            filterable={false}
            freezeWhenExpanded={false}
            getLoadingProps={[Function]}
            getNoDataProps={[Function]}
            getPaginationProps={[Function]}
            getProps={[Function]}
            getResizerProps={[Function]}
            getTableProps={[Function]}
            getTbodyProps={[Function]}
            getTdProps={[Function]}
            getTfootProps={[Function]}
            getTfootTdProps={[Function]}
            getTfootTrProps={[Function]}
            getTheadFilterProps={[Function]}
            getTheadFilterThProps={[Function]}
            getTheadFilterTrProps={[Function]}
            getTheadGroupProps={[Function]}
            getTheadGroupThProps={[Function]}
            getTheadGroupTrProps={[Function]}
            getTheadProps={[Function]}
            getTheadThProps={[Function]}
            getTheadTrProps={[Function]}
            getTrGroupProps={[Function]}
            getTrProps={[Function]}
            groupedByPivotKey="_groupedByPivot"
            indexKey="_index"
            loading={false}
            loadingText="Loading..."
            multiSort={true}
            nestingLevelKey="_nestingLevel"
            nextText="Next"
            noDataText="Numbers Not Generated"
            ofText="of"
            onFetchData={[Function]}
            originalKey="_original"
            pageJumpText="jump to page"
            pageSizeOptions={
              Array [
                5,
                10,
                20,
                25,
                50,
                100,
              ]
            }
            pageText="Page"
            pivotDefaults={Object {}}
            pivotIDKey="_pivotID"
            pivotValKey="_pivotVal"
            previousText="Previous"
            resizable={true}
            resolveData={[Function]}
            rowsSelectorText="rows per page"
            rowsText="rows"
            showPageJump={true}
            showPageSizeOptions={true}
            showPagination={true}
            showPaginationBottom={true}
            showPaginationTop={false}
            sortable={true}
            style={Object {}}
            subRowsKey="_subRows"
          />
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
